src/librustc_codegen_llvm/mir/block.rs

```diff
@@ -457,7 +457,7 @@ impl FunctionCx<'a, 'll, 'tcx> {
                         // we can do what we like. Here, we declare that transmuting
                         // into an uninhabited type is impossible, so anything following
                         // it must be unreachable.
-                        assert_eq!(bx.cx.layout_of(sig.output()).abi, layout::Abi::Uninhabited);
+                        assert!(sig.output().conservative_is_uninhabited(bx.tcx()));
```

Comment:
Wait, why would you replace this? If you want your assert, I'd keep both asserts in.